03-client-api/references/options.yml

```diff
@@ -7,8 +7,8 @@ description:
     for TypeDB Cluster.
```

Comment:
A community member recently observed:
```
Hi, I'm having this weird issue where I get different results on running the same match query inside read & write transactions
```
This turned out to be because they were using reasoning, and the `infer` option only works in a `read` transaction, yet this was not specified in the docs. So in this PR, we add it.